src/daemon/lldpd.c

```diff
@@ -986,8 +986,13 @@ lldpd_dot3_power_pd_pse(struct lldpd_hardware *hardware)
 	TAILQ_FOREACH(port, &hardware->h_rports, p_entries) {
 		if (port->p_hidden_in)
 			continue;
-		if (port->p_protocol != LLDPD_MODE_LLDP)
+#ifdef ENABLE_CDP
+		if (port->p_protocol != LLDPD_MODE_LLDP && port->p_protocol != LLDPD_MODE_CDPV2)
+			continue;
+#else
+		if (port->p_protocol != LLDPD_MODE_LLDP
 			continue;
+#endif
```

Comment:
You don't need this `#ifdef`. If `ENABLE_CDP` is not defined, `port->p_protocol` cannot be equal to `LLDPD_MODE_CDPV2`, so no issue in having a single condition.

Reply:
Done